### ISSUE TITLE
LRP fails with custom rule and repeated call to `attribute`

### ISSUE THREAD

**Issue by jonasricker** [pytorch/captum#1059]
## 🐛 Bug

When LRP is used with a custom propagation rule, calling `attribute` more than a single time fails because the custom rule is removed in https://github.com/pytorch/captum/blob/5543b4a8d256f2fa6d43eb3ac8675235f996c9fe/captum/attr/_core/lrp.py#L382

## To Reproduce

1. Define a model for which no default rule exists and add a custom rule.
1. Set up LRP with the model.
1. Call `attribute` twice.

## Expected behavior

I expect that only those rules which were added by captum are removed with custom rules remaining.

## Environment
Describe the environment used for Captum

 - Captum / PyTorch Version (e.g., 1.0 / 0.4.0): 0.5.0
 - OS (e.g., Linux): Linux
 - How you installed Captum / PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.9.5
 - CUDA/cuDNN version: 11.4
 - GPU models and configuration: 4 RTX 3090
 - Any other relevant information:

**Comment by NarineK** [comment]
@jonasricker, yes, we do indeed remove it because we don't distinguish between the rules that we set and the ones that user sets. We can definitely add a quick fix for that. We will remove the ones that we set internally.

The only problem is that if the custom rule updates any state variables and it gets reused, the user needs to be responsible cleaning those state variables before calling attribute again.

https://github.com/pytorch/captum/blob/master/captum/attr/_core/lrp.py#L395

**Comment by sophmrtn** [comment]
This bug still persists for me as above, and since the master branch has been updated: https://github.com/pytorch/captum/blob/master/captum/attr/_core/lrp.py#L395 doesn't show the quick fix? 

Would someone be able to share an updated line reference to add a quick fix for this?
